code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -4,19 +4,35 @@
 	icon = 'icons/obj/bloodpack.dmi'
 	icon_state = "empty"
 	volume = 200
-
 	var/blood_type = null
+	var/labelled = 0
 
 /obj/item/weapon/reagent_containers/blood/New()
 	..()
 	if(blood_type != null)
-		name = "blood pack [blood_type]"
 		reagents.add_reagent("blood", 200, list("donor"=null,"viruses"=null,"blood_DNA"=null,"blood_type"=blood_type,"resistances"=null,"trace_chem"=null))
 		update_icon()
 
 /obj/item/weapon/reagent_containers/blood/on_reagent_change()
+	if(reagents)
+		var/datum/reagent/blood/B = reagents.has_reagent("blood")
+		if(B && B.data && B.data["blood_type"])
+			blood_type = B.data["blood_type"]
+		else
+			blood_type = null
+	update_pack_name()
```

Comment:
the blood pack's name is automatically updated with the correct blood type in its name, unless someone manually labeled the pack (so traitors can still do traitor things).